src/components/TokenClaimDialog.tsx

```diff
@@ -187,19 +212,29 @@ export default function TokenClaimDialog({
               </Trans>
             </Typography>
           )}
-          {allPoolsWithRewards.map((pool, i, arr) => (
-            <React.Fragment key={pool.poolName}>
-              <ClaimListItem
-                title={pool.poolName}
-                amount={rewardBalances[pool.poolName] || Zero}
-                claimCallback={() => claimPoolReward(pool)}
-                status={
-                  claimsStatuses["allPools"] || claimsStatuses[pool.poolName]
-                }
-              />
-              {i < arr.length - 1 && <Divider key={i} />}
-            </React.Fragment>
-          ))}
+          {gauges.map(
+            (gauge, i, arr) =>
+              userState?.gaugeRewards?.[gauge?.address || ""]?.claimableSDL.gt(
+                Zero,
+              ) &&
+              gauge && (
```

Comment:
shouldn't need to check for gauge existence here. You can just do a `.filter(Boolean)` when you defined `const gauges` so the array will not include any null values.